Plugin~/WebRTCPlugin/Context.cpp

```diff
@@ -127,19 +127,35 @@ namespace WebRTC
         audioDevice = new rtc::RefCountedObject<DummyAudioDevice>();
         nvVideoCapturerUnique = std::make_unique<NvVideoCapturer>();
         nvVideoCapturer = nvVideoCapturerUnique.get();
+
+#if USE_SOFTWARE_ENCODER
```

Comment:
Please use a variable instead of ifdef.
Also, the only difference seems to be 
- webrtc::CreateBuiltinVideoEncoderFactory() -> SoftwareEncoder
- dummyVideoEncoderFactory  -> NvEncoder

So maybe we can only do if for these two ?